bin/dicomwebjs.js

```diff
@@ -1,35 +1,115 @@
+import fs from "fs";
+import path from "path";
+import { Command } from "commander"
+import { dicomweb, instanceDicom, dumpDicom, utils } from "../src/index.js"
 
-const program = new Command();
+const program = new Command()
+
+program.option(
+  "-s, --study <studyInstanceUID>",
+  "Download a specific study instance UID"
+)
+
+program
+  .name("dicomwebjs")
+  .description("dicomwebjs based tools for manipulation of DICOMweb")
+  .version("0.0.1")
+  .option("--seriesUID <seriesUID>", "For a specific seriesUID")
 
 program
-  .name('dicomwebjs')
-  .description('dicomwebjs based tools for manipulation of DICOMweb')
-  .version('0.0.1')
-  .option('--seriesUID <seriesUID>', 'For a specific seriesUID');
-
-program.command('dump')
-  .description('Dump a dicomweb file')
-  .argument('<dicomwebUrl>', 'dicomweb URL or file location')
+  .command("dump")
+  .description("Dump a dicomweb file")
+  .argument("<dicomwebUrl>", "dicomweb URL or file location")
+  .option("--debug", "Set debug level logging")
   .action(async (fileName, options) => {
-    const qido = await dicomweb.readDicomWeb(fileName, options);
+    const qido = await dicomweb.readDicomWeb(fileName, options)
     for (const dict of qido) {
-      dumpDicom({ dict });
+      dumpDicom({ dict })
+    }
+  })
+
+program
+  .command("download")
+  .description("Download dicomweb file(s)")
+  .argument("<dicomwebUrl>", "dicomweb URL to the base DICOMweb service")
+  .option(
+    "-d, --directory <dicomwebdir>",
+    "Download to local DICOMweb directory"
+  )
+  .option("--debug", "Set debug level logging")
+  .action(async (fileName, options) => {
+    utils.logger.setOptions(options)
+    let downloadUrls = [fileName]
+    if (options.study) {
+      downloadUrls = await dicomweb.queryDownloads(fileName, options)
+    }
+    for (const downloadUrl of downloadUrls) {
+      const data = await dicomweb.readDicomWeb(downloadUrl, options)
+      dicomweb.store(downloadUrl, data, options)
+    }
+  })
+
+program
+  .command("download-v2")
+  .description("Download a full DICOM study from a DICOMweb service")
+  .argument("<dicomwebUrl>", "DICOMweb base URL")
+  .option("-S, --study <studyInstanceUID>", "Download a specific study instance UID")
+  .option("-d, --directory <dicomwebdir>", "Download to local directory", "./")
+  .option("--debug", "Enable debug logging")
+  .action(async (dicomwebUrl, options) => {
+    utils.logger.setOptions(options);
+  
+    if (!options.study) {
+      console.error("Error: You must provide a StudyInstanceUID with --study");
+      process.exit(1);
+    }
+  
+    const baseDir = path.join(options.directory, options.study);
+    let downloadDir = baseDir;
+    let counter = 1;
+
+    while (fs.existsSync(downloadDir)) {
+      downloadDir = `${baseDir}_${counter}`;
+      counter++;
     }
+
+    fs.mkdirSync(downloadDir, { recursive: true });
+
+    console.log(`📁 Study will be downloaded to: ${downloadDir}`);
+  
+    const dicomWebConfig = {
```

Comment:
You can have a single function that adds/parses out the dicomWebConfig from the arguments, and a second one that adds/parses the output directory functions as config/output objects.